staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -315,7 +353,27 @@ func (r *crdHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 
 	terminating := apiextensionshelpers.IsCRDConditionTrue(crd, apiextensionsv1.Terminating)
 
-	crdInfo, err := r.getOrCreateServingInfoFor(crd.UID, crd.Name)
+	var crdInfo *crdInfo
+	if utilfeature.DefaultFeatureGate.Enabled(features.StorageVersionAPI) &&
+		utilfeature.DefaultFeatureGate.Enabled(features.APIServerIdentity) &&
+		(requestInfo.Verb == "create" ||
+			requestInfo.Verb == "update" ||
+			requestInfo.Verb == "patch") {
+		var aborted bool
+		crdInfo, err, aborted = r.getOrCreateServingInfoForWrite(crd.UID, crd.Name)
+		// The server aborted storage version update due to the CRD being deleted.
+		// The CR request is forbidden.
```

Comment:
```
// Requests to non-existent resources return 403
```

Are we sure the only case for `abort==true` is CRD deletion?

Reply:
(other than the new "should not happen" panic handling) yes

Reply:
* 403 doesn't seem right
* if the CRD is being deleted, but is pending cleanup of CRs, won't erroring here deadlock?

Reply:
> won't erroring here deadlock?

we don't block delete requests, and I'm not sure if the storage DestroyFunc goes through this code path.

Reply:
we won't deadlock because the CRD finalizer doesn't go through this code path to cleanup CRs